app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/GrantedJourneyController.scala

```diff
@@ -91,6 +100,34 @@ class GrantedJourneyController @Inject()(
     GrantedJourneyController.provideWarningsForm.bindFromRequest.fold(handleInvalidForm, handleValidForm)
   }
 
+  def provideEscalatedToPage(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+    successful(Ok(provideEscalatedToForGrantingPage(provideEscalatedToForm, ViewModel(request.application.name, applicationId))))
+  }
+
+  def provideEscalatedToAction(applicationId: ApplicationId) = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+    def handleValidForm(form: ProvideEscalatedToForm) = {
```

Comment:
I would question the use of EitherT here when we seem to only be handling a Future[Option]]

Wouldn't it be clearer to just map over the future and pattern match the option?

`
  def handleValidForm(form: ProvideEscalatedToForm) = {
      submissionReviewService
        .updateEscalatedTo(form.escalatedTo)(request.submission.id, request.submission.latestInstance.index)
        .map {
          case Some(value)  => Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.GrantedJourneyController.provideWarningsPage(applicationId).url)
          case None         => {
            logger.warn(s"Error granting access for application $applicationId: There was a problem updating the grant warnings on the submission review")
            InternalServerError(errorHandler.internalServerErrorTemplate)
          }
        }
    }
`

Reply:
I agree - fixed